miner/worker.go

```diff
@@ -891,12 +891,17 @@ func (w *worker) commitTransaction(tx *types.Transaction, coinbase common.Addres
 
 	// do not do CCC checks on follower nodes
 	if w.isRunning() {
+		// do gas limit check up-front and do not run CCC if it fails
+		if w.current.gasPool.Gas() < tx.Gas() {
```

Comment:
It already checked at [preCheck](https://github.com/scroll-tech/go-ethereum/blob/417dd0282743a10d49699c2318487887ae0aa0c4/core/state_transition.go#L340), Is it necessary to check again?

Reply:
We need to check it before calling `w.circuitCapacityChecker.ApplyTransaction`.